tests/reporter/logfilter.go

```diff
@@ -0,0 +1,44 @@
+package reporter
+
+import (
+	"bufio"
+	"bytes"
+	"fmt"
+	"os"
+	"regexp"
+	"strings"
+	"time"
+)
+
+// filterBySinceTimestamp filters lines by matching timestamp where each line timestamp is after 'since' timestamp.
+// Each line timestamp is parsed by the 'parseFn'.
+func filterBySinceTimestamp(content string, timestampRegex *regexp.Regexp, since time.Time, parseFn func(string) (time.Time, error)) string {
+	defer func() {
+		if r := recover(); r != nil {
+			fmt.Fprintf(os.Stderr, "\nfilterBySinceTimestamp: panic: %v", r)
+		}
+	}()
+
+	filtered := strings.Builder{}
```

Comment:
nit: maybe add a minimal happy path unit test to make sure we don't lose important stuff here?